src/lib/templateengine.js

```diff
@@ -1371,6 +1376,14 @@ function TemplateEngine( undefined ) {
 
   this.create_pages = function(page, path, flavour, type) {
     var creator = thisTemplateEngine.design.getCreator(page.nodeName);
+    
+    thisTemplateEngine.callbacks[ path + '_' ] = {
+      exitingPageChange: [],// called when the current page is left
+      beforePageChange: [], // called as soon as a page change is known
+      duringPageChange: [], // called when the page is theoretical visible, i.e. "display:none" is removed - CSS calculations shoud work now
+      afterPageChange: []   // called together with the global event when the transition is finished
```

Comment:
The idea is that communication to get the necessary content can already be started by `beforePageChange`. So this is called as soon as the change is known.  
The `duringPageChange` comes shortly afterwards. Here the `display:none` is already removed so the CSS measurements should work now. This also means that a reflow has happend. This could take some time - not much, but during that time the server could already process the response.  
The `afterPageChange` happens once the page is finally there as it should be. This means that the page transision / animation time has passed.

In the past only the last event was available - and that "only" via a "slow" global event. The `afterPageChange` uses a direct callback, i.e. is quite lightweight.  
As it comes quite late it's definitely a different time point than the other callbacks.

`beforePageChange` and `duringPageChange` are really close in time. So they could be unified to the `duringPageChange` (a unification to `beforePageChange` can not work as CSS measurements won't work at that time).  
But I recon the little speed advantage of kicking of the communication before and parallel to the reflow is worth the ~5 lines of additional code.